### PR TITLE
feat: add grpc support and update communicator factory

### PR DESCRIPTION
**Pull request by DanielRuthvik**

### CODE REVIEW

File: Networking/Communication/ICommunicator.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ public string Start(string serverIP = null,
         /// The socket which is connected to the client.
         /// </param>
         /// <returns> void </returns>
-        public void AddClient(string clientId, TcpClient socket);
+        public void AddClient(string clientId, TcpClient socket, string ip = null, string port = null);
```
Comment (ChitreshN) [inline Networking/Communication/ICommunicator.cs:47]: inform dashboard team about these changes

File: Networking/INotificationHandler.cs [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+    {
+        /// <summary>
+        /// Called when data of a particular module appears in the receiving queue
+        /// </summary>
+        public void OnDataReceived(string serializedData);
+
+        /// <summary>
+        /// Called on the server when a new client joins
+        /// </summary>
+        public void OnClientJoined(TcpClient socket, string ip = null, string port = null)
```
Comment (ChitreshN) [inline Networking/INotificationHandler.cs:15]: inform about these changes

File: Networking/Protos/server.proto [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+service Server{
+	rpc connect (connectRequest) returns (connectResponse); // server that helps in storing the ip and port of the client in a map
+	rpc serverReceive (sendRequest) returns (sendResponse); // this server is called when send function is called on the client side 
+    rpc disconnect (disconnectRequest) returns (disconnectResponse); // this server is called when the client disconnects
+	rpc updateCounter (updateCounterRequest) returns (updateCounterResponse);
+	rpc updateCounter2 (updateCounterRequest) returns (updateCounterResponse);
+}
+
+message disconnectRequest{
```
Comment (ChitreshN) [inline Networking/Protos/server.proto:15]: add more comments explaining the message structure

File: Networking/Protos/server.proto [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	string clientId = 1;
+}
+
+message disconnectResponse{
+
+}
+
+message updateCounterRequest{
```
Comment (ChitreshN) [inline Networking/Protos/server.proto:23]: huh?

**Review by ChitreshN — APPROVED** [review]

**Review by Tlakshmisagar — COMMENTED** [review]
some files are missing comments

**Pull request merged**